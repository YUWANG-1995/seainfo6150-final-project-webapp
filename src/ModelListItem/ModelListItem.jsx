import React from "react";
import styles from "./ModelListItem.module.css";

const ModelListItem = (props) => {
  return (
    <div className={styles.container}>
      <img className={styles.image} src={props.model.image} alt="it's a model" />
      <p className={styles.modelName}>Model: {props.model.modelId}</p>
    </div>
  );
}

export default ModelListItem;
