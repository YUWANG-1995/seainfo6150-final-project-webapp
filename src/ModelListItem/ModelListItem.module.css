.container{
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
}

.image{
  width: 500px;
  height: 300px;
}

.modelName{
  font-family: 'Roboto', sans-serif;
  font-weight: bold;
  font-size: 1.5em;
  color: #000000;
}

p.modelName:hover{
  color: red;
}
