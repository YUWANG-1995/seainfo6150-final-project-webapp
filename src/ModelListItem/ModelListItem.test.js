import React from "react";
import { render } from "@testing-library/react";
import ModelListItem from "./ModelListItem.jsx";

it("renders the ModelList Component Correctly", ()=> {
  const test = {
      brand: 'BMW',
      model: "X1"
  }

  const { container } = render(
    <ModelListItem
      brand = {test.brand} model={test.model}
    />
  );
  expect(container).toMatchSnapshot();
})
