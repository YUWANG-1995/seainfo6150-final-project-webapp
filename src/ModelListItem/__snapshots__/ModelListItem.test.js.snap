// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ModelList Component Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="it's a model"
      class="image"
    />
    <p
      class="modelName"
    >
      Model: 
    </p>
  </div>
</div>
`;
